apps/studio/components/interfaces/DiskManagement/DiskManagement.utils.ts

```diff
@@ -109,3 +147,195 @@ export const calculateThroughputPrice = ({
+    computeOptions.unshift({
+      identifier: 'ci_micro',
+      name: 'Micro',
+      price_description: '$0.01344/hour (~$10/month)',
+      price: 0.01344,
+      price_interval: 'hourly',
+      price_type: 'usage',
+      meta: {
+        cpu_cores: INSTANCE_MICRO_SPECS.cpu_cores,
+        cpu_dedicated: INSTANCE_MICRO_SPECS.cpu_dedicated,
+        memory_gb: INSTANCE_MICRO_SPECS.memory_gb,
+        baseline_disk_io_mbs: INSTANCE_MICRO_SPECS.baseline_disk_io_mbs,
+        max_disk_io_mbs: INSTANCE_MICRO_SPECS.max_disk_io_mbs,
+        connections_direct: INSTANCE_MICRO_SPECS.connections_direct,
+        connections_pooler: INSTANCE_MICRO_SPECS.connections_pooler,
+      } as ProjectAddonVariantMeta,
+    })
+  }
+
+  computeOptions.unshift({
+    identifier: 'ci_nano',
+    name: 'Nano',
+    price_description: '$0/hour (~$0/month)',
+    price: 0,
+    price_interval: 'hourly',
+    price_type: 'usage',
+    // @ts-ignore API types it as Record<string, never>
+    meta: {
+      cpu_cores: INSTANCE_NANO_SPECS.cpu_cores,
+      cpu_dedicated: INSTANCE_NANO_SPECS.cpu_dedicated,
+      memory_gb: INSTANCE_NANO_SPECS.memory_gb,
+      baseline_disk_io_mbs: INSTANCE_NANO_SPECS.baseline_disk_io_mbs,
+      max_disk_io_mbs: INSTANCE_NANO_SPECS.max_disk_io_mbs,
+      connections_direct: INSTANCE_NANO_SPECS.connections_direct,
+      connections_pooler: INSTANCE_NANO_SPECS.connections_pooler,
+    } as ProjectAddonVariantMeta,
+  })
+
+  return computeOptions
+}
+
+export const calculateMaxIopsAllowedForDiskSizeWithGp3 = (totalSize: number) => {
+  if (!totalSize) throw Error('Total size is required')
+  return Math.min(3000 * totalSize, 16000)
+}
+
+export const calculateDiskSizeRequiredForIopsWithGp3 = (iops: number) => {
+  if (!iops) throw Error('IOPS is required')
+  return Math.max(1, Math.ceil(iops / 500))
+}
+
+export const calculateMaxIopsAllowedForDiskSizeWithio2 = (totalSize: number) => {
+  if (!totalSize) throw Error('Total size is required')
+  return Math.min(500 * totalSize, 256000)
+}
+
+export const calculateDiskSizeRequiredForIopsWithIo2 = (iops: number) => {
+  if (!iops) throw Error('IOPS is required')
+  return Math.max(4, Math.ceil(iops / 1000))
+}
+
+export const calculateMaxThroughput = (iops: number) => {
+  if (!iops) throw Error('IOPS is required')
+  return Math.min(0.256 * iops, 1000)
+}
+
+export const calculateIopsRequiredForThroughput = (throughput: number) => {
+  if (!throughput) throw Error('Throughput is required')
+  return Math.max(125, Math.ceil(throughput / 0.256))
+}
+
+export const calculateMaxIopsAllowedForComputeSize = (computeSize: string): number => {
+  if (!computeSize) throw Error('Compute size is required')
+  return COMPUTE_BASELINE_IOPS[computeSize as keyof typeof COMPUTE_BASELINE_IOPS] || 0
+}
+
+export const calculateComputeSizeRequiredForIops = (
+  iops: number
+): ComputeInstanceAddonVariantId => {
+  if (iops === undefined || iops === null) {
+    throw new Error('IOPS is required')
+  }
```

Comment:
this check seems unnecessary - will never run into this block as the props are declared to be number type and non optional